### PR TITLE
Make improvements to the UI again

### PR DESCRIPTION
**Pull request by Senexis**
## Notes
This took a while. This makes the difference checker much more usable and finished-looking. Tested this for a while and all seems good, although there's still some issues such as for some reason different browsers showing different changes at the same version (between Chrome and Firefox, at least). For this reason the whole "site under construction" message still exists.

This sort of thing is more interesting than I thought (to me at least), allows you to see the exact changes between builds. I just wish the builds were described with something other than just a hash and timestamp. Maybe someday.

_- This shouldn't have any sneaky file removes. I know because I checked this time._ 😄 

### Difference checker page
Features:
- Use the style of the drops page.
- Allow the user to select different available builds to compare between.
- Allow objects and arrays to be closed.

Fixes:
- Fix array moves being listed as edits by (thoroughly) sorting the different arrays by name instead of id.
- Fix missing context by always displaying the item, objective, or reward name when something within it is edited.
- Fix the page not indicating it's loading data.
- Don't display commas on the last elements.

### Drops page
Changes:
- Add a message when the user doesn't have JavaScript enabled.
- Improve the styling of the loading screens.
- Add a link to the compare versions page since that's now usable.
- Align the column icons to the right, as that's usually how they are aligned in tables.

Fixes:
- Allow the rarity column to be sorted (it sorts by chance, but don't tell anyone).
- Fix the columns becoming broken on smaller screens by enforcing a minimum width (scrolling is better than jumbled text).
- Enforce "/search" in the URL.

### CODE REVIEW

**Comment by Senexis** [comment]
Since they aren't at all related, I decided against that on purpose. I understand that the other PR would spark some discussion as to semantics, but this is just a general UI improvements.

**Comment by TobiTenno** [comment]
seems fine, @atomicptr i don't know if enforcing `/search` will break open search.

**Comment by Senexis** [comment]
It shouldn't do as that is expected behavior. :)

Viewable at:
- https://senexis.github.io/warframe-drop-data/
- https://senexis.github.io/warframe-drop-data/diff.html

**Comment by TobiTenno** [comment]
I'm not saying it is, I just want to make sure they are good with it.

**Comment by TobiTenno** [comment]
If you're adding a view to the diff, wanna make sure they're sorted by something so that the individual sections appear close to where they were moved or something similar?

**Comment by Senexis** [comment]
That's already happening to avoid item moves being shown like they are on the "live" version, so what do you mean exactly?

**Comment by Senexis** [comment]
Finally got around to handling the missionRewards. It was previously missing, but it's now fully implemented and being sorted. This entire thing is actually pretty close to being fully finished and no longer "(WORK IN PROGRESS!!!)". 😏

File: diff.html [hunk 1]
```diff
@@ -344,15 +344,33 @@ <h5>Warframe Drop Data - Compare Versions</h5>
...
-                else if (build[key][sub]["rewards"] && Array.isArray(build[key][sub]["rewards"].A)) { newArray[key][sub]["rewards"].A = build[key][sub]["rewards"].A.sort(sortItemName) }
-                else if (build[key][sub]["rewards"] && Array.isArray(build[key][sub]["rewards"].B)) { newArray[key][sub]["rewards"].B = build[key][sub]["rewards"].B.sort(sortItemName) }
-                else if (build[key][sub]["rewards"] && Array.isArray(build[key][sub]["rewards"].C)) { newArray[key][sub]["rewards"].C = build[key][sub]["rewards"].C.sort(sortItemName) }
+                else if (build[key][sub]["rewards"] && Array.isArray(build[key][sub]["rewards"]["A"]) && Array.isArray(build[key][sub]["rewards"]["B"]) && Array.isArray(build[key][sub]["rewards"]["C"])) {
+                    newArray[key][sub]["rewards"]["A"] = build[key][sub]["rewards"]["A"].sort(sortItemName)
+                    newArray[key][sub]["rewards"]["B"] = build[key][sub]["rewards"]["B"].sort(sortItemName)
+                    newArray[key][sub]["rewards"]["C"] = build[key][sub]["rewards"]["C"].sort(sortItemName)
+                }
+                else {
+                    if (key === "missionRewards") {
```
Comment (TobiTenno) [inline diff.html]: this should be combined with the above line to make it
```js
} else if (key === "missionRewards") {
```

Comment (Senexis) [reply diff.html]: Good observation. However, I'm still getting some data in there that's not been handed, so I'm still not completely done with that construction. As soon as I've handled the last remaining data, I will convert it for consistency.

Comment (Senexis) [reply diff.html]: All done. Changed it for consistency.

**Comment by Senexis** [comment]
If there's nothing else you guys want changed, I would now regard this as done. 🎉

File: index.html [hunk 2]
```diff
@@ -104,7 +118,7 @@
             <div>
                 <img src="misc/logo.png" height="64">
                 <h5>Warframe Drop Data</h5>
-                <a target="_blank" href="https://github.com/WFCD/warframe-drop-data#api-endpoints">API Documentation</a> | <a target="_blank" href="https://github.com/WFCD/warframe-drop-data">Github</a>
+                <a href="https://drops.warframestat.us/diff.html">Compare Versions</a> | <a target="_blank" href="https://github.com/WFCD/warframe-drop-data#api-endpoints">API Documentation</a> | <a target="_blank" href="https://github.com/WFCD/warframe-drop-data">Github</a>
```
Comment (TobiTenno) [inline index.html]: link to `./diff.html` instead should work better for the compare versions
that way it's domain agnostic

Comment (Senexis) [reply index.html]: Done.

**Comment by atomicptr** [comment]
One minor complaint, in the diff viewer instead of using the US date layout of MM/DD/YYYY HH:mm I'd much rather use ISO 8601 date format which is YYYY-MM-DD HH:mm (also in 24h).

Only complaint I've found so far tho :>

File: diff.html [hunk 3]
```diff
@@ -1,33 +1,434 @@
...
+<section>
+    <div class="container">
+        <div class="header">
+            <div>
+                <img src="misc/logo.png" height="64">
+                <h5>Warframe Drop Data - Compare Versions</h5>
+                <a href="http://drops.warframestat.us/">Drop Data</a> | <a target="_blank" href="https://github.com/WFCD/warframe-drop-data#api-endpoints">API Documentation</a> | <a target="_blank" href="https://github.com/WFCD/warframe-drop-data">Github</a>
+            </div>
+            <div id="builds-selection">
+                <p>Compare <select id="builds-old" disabled="disabled"><option value="">Please select a build...</option></select> with <select id="builds-new" disabled="disabled"><option value="">Please select a build...</option></select></p>
```
Comment (atomicptr) [inline diff.html]: The default/empty option should also have an "disabled" attribute

**Comment by atomicptr** [comment]
The drops page changes look fine, though since you made the rarity sortable, wouldn't it make sense to merge the two? Insteadof having 2 columns "Rarity", "Chance" this could just be "Rarity" and display something like "Rare (5.60%)"

**Comment by Senexis** [comment]
@atomicptr This is up to your browser, since it's using toLocaleString(). If I'm guessing, you're like me and have it set to English (US) which causes it to be displayed in the US format. If you set your browser to something else, it will reflect it.

**Comment by Senexis** [comment]
It would make sense to merge the two, yeah. Don't know why I decided to keep the two columns as they don't make sense. Would suggest making it "5% (Rare)" though as the focus should be the percentage. That's what people are more interested in, after all. "Rare" can be found in the codex.

**Comment by atomicptr** [comment]
Either way is fine for me :).

File: diff.html (hunk 3 above)
Comment (Senexis) [reply diff.html]: Added.

**Comment by Senexis** [comment]
All done. Decided to go the "Uncommon (20%)" route since it looks better.

**Comment by atomicptr** [comment]
Looks good, only one issue with the diff page left:

> Also changing the versions doesn't seem to work at all times, sometimes it does with loading screen and all and sometimes nothing happens.

**Comment by Senexis** [comment]
Care to elaborate on that issue? If it's been a while since that occured, it might be fixed in 488b6fb3a175db77354381835063d8b4e57d6c94.

**Comment by atomicptr** [comment]
Hm, might have been the cache it works now. The thing with the "the previous version has to be before blabla" one is a bit weird UX wise but it's fine will merge, ty for your work :).

**Comment by Senexis** [comment]
Yeah I wasn't sure whether to allow it or not. Allowing it could be confusing, so I'd thought its better to have it "nope" than to cause confusion because you're comparing a newer version to an older version.

**Pull request merged**